pkg/api/controllers/transaction_controller.go

```diff
@@ -59,9 +63,24 @@ func (ctl *TransactionController) GetTransactions(c *gin.Context) {
 		}
 	}
 
+	after := c.Query("after")
+	if c.Query("pagination_token") != "" {
+		res, err := base64.RawURLEncoding.DecodeString(c.Query("pagination_token"))
+		if err != nil {
+			ResponseError(c, ledger.NewValidationError("invalid query value 'pagination_token'"))
+			return
+		}
+		t := sqlstorage.PaginationToken{}
+		if err = json.Unmarshal(res, &t); err != nil {
+			ResponseError(c, ledger.NewValidationError("invalid query value 'pagination_token'"))
+			return
+		}
+		after = strconv.FormatUint(t.ID, 10)
+	}
+
 	cursor, err := l.(*ledger.Ledger).GetTransactions(
```

Comment:
Use pagination_token with others query params (reference, account or source) can lead to unexpected behavior.
Also, client side, it is more complicated to deal with the pagination if we have to store all theses variables.
In the best of the world, those parameters are defined at the first query and never retransmitted.
So i think it should be stored inside the pagination_token and other query params should be ignored (or trigger an error) if specified.

Reply:
Ok thanks! I agree 👍